src/Application/Items/Commands/ReforgeUpgradedUserItemCommand.cs

```diff
@@ -0,0 +1,101 @@
+    {
+        private static readonly ILogger Logger = LoggerFactory.CreateLogger<ReforgeUpgradedUserItemCommand>();
+
+        private readonly ICrpgDbContext _db;
+        private readonly IMapper _mapper;
+        private readonly IActivityLogService _activityLogService;
+        private readonly Constants _constants;
+
+        public Handler(ICrpgDbContext db, IMapper mapper, IActivityLogService activityLogService, Constants constants)
+        {
+            _db = db;
+            _mapper = mapper;
+            _activityLogService = activityLogService;
+            _constants = constants;
+        }
+
+        public async Task<Result<UserItemViewModel>> Handle(ReforgeUpgradedUserItemCommand req, CancellationToken cancellationToken)
+        {
+            var user = await _db.Users
+                .AsSplitQuery()
+                .Include(u => u.Items).ThenInclude(ui => ui.Item)
+                .FirstOrDefaultAsync(u => u.Id == req.UserId, cancellationToken);
+
+            if (user == null)
+            {
+                return new(CommonErrors.UserNotFound(req.UserId));
+            }
+
+            var userItemToReforge = user.Items
+                .FirstOrDefault(ui => ui.Id == req.UserItemId);
+
+            if (userItemToReforge == null)
+            {
+                return new(CommonErrors.UserItemNotFound(req.UserItemId));
+            }
+
+            if (userItemToReforge.IsBroken)
+            {
+                return new(CommonErrors.ItemBroken(userItemToReforge.ItemId));
+            }
+
+            if (userItemToReforge.Item!.Rank == 0)
+            {
+                return new(CommonErrors.ItemNotReforgeable(userItemToReforge.ItemId));
+            }
+
+            int reforgePrice = (int)_constants.ItemReforgeCostPerRank[userItemToReforge.Item.Rank];
+
+            if (user.Gold < reforgePrice)
+            {
+                return new(CommonErrors.NotEnoughGold(reforgePrice, user.Gold));
+            }
+
+            Item? baseItem = await _db.Items
+                .FirstOrDefaultAsync(
+                    i => i.BaseId == userItemToReforge.Item!.BaseId && i.Rank == 0,
+                    cancellationToken);
+
+            if (baseItem == null)
+            {
+                return new(CommonErrors.ItemNotReforgeable(userItemToReforge.ItemId));
+            }
+
+            if (user.Items.Any(ui => ui.Item!.Id == baseItem.Id))
+            {
+                return new(CommonErrors.ItemAlreadyOwned(baseItem.Id));
+            }
+
+            _db.ActivityLogs.Add(_activityLogService.CreateItemReforgedLog(user.Id, userItemToReforge.ItemId, userItemToReforge.Item.Rank));
+            user.HeirloomPoints += userItemToReforge.Item.Rank;
+            user.Gold -= reforgePrice;
+            userItemToReforge.Item = baseItem;
+
+            await _db.SaveChangesAsync(cancellationToken);
+
+            Logger.LogInformation("User '{0}' has Reforged item '{1}'", req.UserId, req.UserItemId);
```

Comment:
Reforged -> reforged

Reply:
Also can you use the item id instead of the user item id

Reply:
done